includes/rest-api/class-amp-rest-fonts-controller.php

```diff
@@ -0,0 +1,242 @@
+	 *
+	 * @param Object          $font Font object.
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $font, $request ) {
+		$fields = $this->get_fields_for_response( $request );
+		$schema = $this->get_item_schema();
+		$data   = [];
+
+		if ( in_array( 'name', $fields, true ) ) {
+			$data['name'] = $font['name'];
+		}
+
+		if ( in_array( 'slug', $fields, true ) ) {
+			$data['slug'] = $font['slug'];
+		}
+
+		if ( in_array( 'handle', $fields, true ) ) {
+			$data['handle'] = isset( $font['handle'] ) ? $font['handle'] : '';
+		}
+
+		if ( in_array( 'src', $fields, true ) ) {
+			$data['src'] = isset( $font['src'] ) ? $font['src'] : $schema['properties']['src']['default'];
+		}
+
+		if ( in_array( 'fallbacks', $fields, true ) ) {
+			$data['fallbacks'] = isset( $font['fallbacks'] ) ? (array) $font['fallbacks'] : $schema['properties']['fallbacks']['default'];
+		}
+
+		if ( in_array( 'weights', $fields, true ) ) {
+			$data['weights'] = isset( $font['weights'] ) ? (array) $font['weights'] : $schema['properties']['weights']['default'];
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		/**
+		 * Filters a font returned from the API.
+		 *
+		 * Allows modification of the font right before it is returned.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param Object $font The original font object.
+		 * @param WP_REST_Request $request Request used to generate the response.
+		 */
+		return apply_filters( 'rest_prepare_font', $response, $font, $request );
+	}
+
+
+	/**
+	 * Checks if a given request has access to get fonts.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 *
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) {
+		return current_user_can( 'edit_posts' );
+	}
+
+	/**
+	 * Retrieves the font' schema, conforming to JSON Schema.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+		$schema       = [
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'font',
+			'type'       => 'object',
+			'properties' => [
+				'name'      => [
+					'description' => __( 'The title for the status.', 'amp' ),
+					'type'        => 'string',
+					'context'     => [ 'embed', 'view', 'edit' ],
+					'readonly'    => true,
+				],
+				'slug'      => [
+					'description' => __( 'An alphanumeric identifier for the status.', 'amp' ),
```

Comment:
Should this be for `font` and not `status`?

Reply:
Changed in https://github.com/ampproject/amp-wp/pull/3998/commits/7a80cd18b87d6fab2c775a0f037d366996fc49cb